samples/sdl_gamecontroller/main.c

```diff
@@ -0,0 +1,67 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include <assert.h>
+#include <SDL.h>
+#include <stdbool.h>
+
+int main() {
+  int ret = pb_init();
+  if (ret != 0) {
+    XSleep(2000);
+    return -1;
+  }
+
+  pb_show_debug_screen();
+
+  SDL_InitSubSystem(SDL_INIT_GAMECONTROLLER);
+
+  int gameControllerIndex = 0;
+  assert(SDL_IsGameController(gameControllerIndex));
+  SDL_GameController *gameController = SDL_GameControllerOpen(gameControllerIndex);
+  if (gameController == NULL) {
+    debugPrint("Could not open GameController %i, exiting...", gameControllerIndex);
+    SDL_QuitSubSystem(SDL_INIT_GAMECONTROLLER);
+    XSleep(2000);
+    return -1;
+  }
+
+  debugPrint("GameController name: %s\n", SDL_GameControllerName(gameController));
+  debugPrint("Is GameController  : %s\n", SDL_IsGameController(gameControllerIndex)?"Yes":"No");
+  debugPrint("GameController Map : %s\n", SDL_GameControllerMapping(gameController));
+  debugPrint("\n");
+
+  while (true) {
+    // Fetch current GameController state
+    SDL_GameControllerUpdate();
+    debugClearScreen();
+    debugPrint("%s:%i      ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_A),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_A));
+    debugPrint("%s:%i      ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_B),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_B));
+    debugPrint("%s:%i      ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_X),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_X));
+    debugPrint("%s:%i\n",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_Y),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_Y));
+    debugPrint("%s:%i                 ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_BACK),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_BACK));
+    debugPrint("%s:%i\n",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_START),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_START));
+    debugPrint("%s:%i       ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_LEFTSTICK),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_LEFTSTICK));
+    debugPrint("%s:%i\n",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_RIGHTSTICK),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_RIGHTSTICK));
+    debugPrint("%s:%i ",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_LEFTSHOULDER),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_LEFTSHOULDER));
+    debugPrint("%s:%i\n\n",SDL_GameControllerGetStringForButton(SDL_CONTROLLER_BUTTON_RIGHTSHOULDER),SDL_GameControllerGetButton(gameController, SDL_CONTROLLER_BUTTON_RIGHTSHOULDER));
```

Comment:
These "\n\n" should probably be move into separate `debugPrint` to be more visible.

Reply:
Consider it done. :+1: